contracts/hydro/src/testing.rs

```diff
@@ -1470,6 +1470,157 @@ fn switch_vote_between_short_and_long_props_test() {
     );
 }
 
+// Test case:
+//      1. User locks tokens and votes for some proposal with longer deployment duration
+//      2. User locks more tokens, which automatically votes for proposal from step #1
+//      3. When the next round starts, user tries to vote for some proposal with the lockup created in step #2
+#[test]
+fn disable_voting_in_next_round_with_auto_voted_lock_test() {
+    let user_address = "addr0000";
+    let user_token = Coin::new(1000u64, IBC_DENOM_1.to_string());
+
+    let grpc_query = denom_trace_grpc_query_mock(
+        "transfer/channel-0".to_string(),
+        HashMap::from([(IBC_DENOM_1.to_string(), VALIDATOR_1_LST_DENOM_1.to_string())]),
+    );
+    let (mut deps, mut env) = (mock_dependencies(grpc_query), mock_env());
+    let info = get_message_info(&deps.api, user_address, &[user_token.clone()]);
+    let mut instantiate_msg = get_default_instantiate_msg(&deps.api);
+    instantiate_msg.round_length = ONE_MONTH_IN_NANO_SECONDS;
+
+    let res = instantiate(
+        deps.as_mut(),
+        env.clone(),
+        info.clone(),
+        instantiate_msg.clone(),
+    );
+    assert!(res.is_ok());
+
+    let current_round_id = 0;
+    let tranche_id = 1;
+
+    let first_proposal_id = 0;
+    let second_proposal_id = 1;
+
+    let first_lock_id = 0;
+    let second_lock_id = 1;
+
+    let res = set_validator_infos_for_round(
+        &mut deps.storage,
+        current_round_id,
+        vec![VALIDATOR_1.to_string()],
+    );
+    assert!(res.is_ok());
+
+    env.block.time = env.block.time.plus_hours(12);
```

Comment:
not needed, right?

Reply:
Yes not mandatory. I added it to simulate lock not being created immediately the same second that the round starts. Will remove it.